src/control/destination/background/mod.rs

```diff
@@ -51,18 +51,30 @@ type UpdateRx<T> = Receiver<PbUpdate, T>;
 /// provided authority to a set of addresses, and ensures that resolution updates are
 /// propagated to all requesters.
 struct Background<T: HttpService<ResponseBody = RecvBody>> {
+    cfg: QueryCfg,
     dns_resolver: dns::Resolver,
-    namespaces: Namespaces,
-    destinations: HashMap<DnsNameAndPort, DestinationSet<T>>,
-    /// A queue of authorities that need to be reconnected.
-    reconnects: VecDeque<DnsNameAndPort>,
+    dsts: DestinationCache<T>,
     /// The Destination.Get RPC client service.
     /// Each poll, records whether the rpc service was till ready.
     rpc_ready: bool,
     /// A receiver of new watch requests.
     request_rx: mpsc::UnboundedReceiver<ResolveRequest>,
 }
 
+/// Holds the currently active `DestinationSet`s and a list of any destinations
+/// which require reconnects.
+struct DestinationCache<T: HttpService<ResponseBody = RecvBody>> {
+    destinations: HashMap<DnsNameAndPort, DestinationSet<T>>,
+    /// A queue of authorities that need to be reconnected.
+    reconnects: VecDeque<DnsNameAndPort>,
+}
+
+/// The configurationn necessary to create a new Destination service
+/// query.
+struct QueryCfg {
```

Comment:
I'm not particularly fond of the names "DestinationCache" and "QueryCfg" --- these are simply the best names I could think of that weren't already being used for something else in this module. 

If anyone can think of more descriptive names, I'd welcome suggestions!